### PR TITLE
fix(solo): introduce runStake attMaker to wallet admin facet

### LINKED ISSUE #4263: Make the wallet support getRUN / BLD Boost
## What is the Problem Being Solved?

For the `getRUN` protocol, dapps (e.g. the treasury) need to be able to propose offers for the wallet to lock and unlock various amounts of BLD, and borrow/repay various amounts of RUN. The problem is that with `getRUN`, the wallet needs to get an attestation for the `give` amount in the offers ([ref](https://github.com/Agoric/agoric-sdk/blob/dc-run-loc/packages/treasury/test/test-runLoC.js#L258)). Also, the BLD purse cannot be selected since it always must be the virtual BLD purse that can contain locked tokens. Due to these differences from a regular offer proposal, the wallet needs a way to know when to handle this separate case.

## Description of the Design

Add a new method to the [wallet bridge API](https://agoric.com/documentation/guides/wallet/api.html#walletbridge-api-commands) called `addGetRUNProposal` that takes a give/want amount of BLD/RUN. It should show an offer in the UI similar to regular offers, but when accepted it should get an attestation and do all of the required steps for `getRUN` (see ref above).

Note this doesn't include the other backend work required to support `makeAttestations` from the wallet, just the API that the dapps will use.

## Security Considerations

The UI should make it clear that tokens are being locked rather than "given" to the contract. It should also reflect the locked amount in the purse.

## Test Plan

Unit testing and e2e testing with the treasury dapp

### PR DESCRIPTION
**Pull request by dckc**
closes #4263

## Description

This addresses a gap in the runStake user flows where a
user would have to use the REPL to do:

 E(E(home.wallet).getAdminFacet()).resolveAttMaker(home.attMaker)

We use setPresences to detect the arrival of attMaker on the chain
bundle (decentralObjects) and setWallet to detect the arrival of the
wallet. When both are available, we introduce them with
resolveAttMaker.

### Security Considerations

It's by design that a user's wallet has the power to place liens on their stake.

### Documentation Considerations

reduces the need for special "now do this in the REPL" step for runStake.

### Testing Considerations

@samsiegart please verify that this works with run.agoric.net.

I had some trouble getting the synchronization right. @michaelfig and I found that by adding an await ... somewhere... (in `registerWallet()`?) it would work. Then I stashed it and came back to it, and that await was gone, but this didn't fail like it seemed to earlier.

### CODE REVIEW

File: packages/solo/src/vat-http.js [hunk 1]
```diff
@@ -97,6 +104,14 @@ export function buildRootObject(vatPowers) {
     urlToHandler.set(url, commandHandler);
   }
 
+  const [resolveWallet, resolveAttMaker] = introduceWith(
+    async (wallet, attMaker) => {
+      const walletAdmin = await E(wallet).getAdminFacet();
+      // console.debug('introduce', { wallet, walletAdmin, attMaker });
+      E(walletAdmin).resolveAttMaker(attMaker);
```
Comment (samsiegart) [inline packages/solo/src/vat-http.js]: Ah didn't expect this method to be useful still, cool!

File: packages/solo/src/vat-http.js [hunk 2]
```diff
@@ -3,8 +3,15 @@ import { E } from '@endo/eventual-send';
 import { Far } from '@endo/marshal';
 import { assert, details as X } from '@agoric/assert';
 import { getReplHandler } from '@agoric/vats/src/repl.js';
+import { makePromiseKit } from '@endo/promise-kit';
 import { getCapTPHandler } from './captp.js';
 
+const introduceWith = f => {
```
Comment (samsiegart) [inline packages/solo/src/vat-http.js]: Nit: This function seems a little mysterious on its own, maybe name it something more specific to attMaker or just do it inline.

**Review by samsiegart — APPROVED** [review]

File: packages/solo/src/vat-http.js (hunk 1 above)
Comment (dckc) [reply packages/solo/src/vat-http.js]: I was hoping / planning to not use it, but I couldn't figure out an alternative.

**Comment by samsiegart** [comment]
I just saw that I need to test this manually, will do so then re-review

**Review by samsiegart — APPROVED** [review]
I tested this with a local chain and it seemed to work

**Pull request merged**